### PR TITLE
Sync zed into 2023.1

### PR DESCRIPTION
**Pull request by jovial**

### CODE REVIEW

File: etc/kayobe/pulp.yml [hunk 1]
```diff
@@ -523,8 +523,8 @@ stackhpc_pulp_images_kolla:
 
 # List of images for each base distribution which should not/cannot be built.
 stackhpc_kolla_unbuildable_images:
-  ubuntu: []
-  rocky: []
+  ubuntu-jammy: []
```
Comment (jovial) [inline etc/kayobe/pulp.yml:526]: Hmm, perhaps we didn't want this change.

File: tools/kolla-images.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+    "neutron_tls_proxy": "haproxy",
+    "nova_compute_ironic": "nova",
+    "redis_sentinel": "openstack",
+    "swift_object_expirer": "swift",
+    "tgtd": "iscsi",
+}
+
+# List of supported base distributions and versions.
+SUPPORTED_BASE_DISTROS = [
+    "rocky-9",
```
Comment (jovial) [inline tools/kolla-images.py:88]: rocky and ubuntu

File: tools/kolla-images.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+    subparsers.add_parser("list-tags", help="List tags for each image based on kolla-image-tags.yml")
+
+    subparsers.add_parser("list-tag-vars", help="List Kolla Ansible tag variables")
+
+    return parser.parse_args()
+
+def get_distro_regex_map(openstack_release):
+    distro_to_regex = {
+        "rocky": re.compile(f"^{openstack_release}-rocky-9-[\d]{{8}}T[\d]{{6}}$"),
```
Comment (jovial) [inline tools/kolla-images.py]: Not sure if this is the cleanest solution but there is mismatch between the kolla_base_distro and the tag we use for the docker image (which contains the release version)...

File: tools/kolla-images.py [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+            value = value.strip()
+            for prefix in ("stable/", "unmaintained/"):
+                if value.startswith(prefix):
+                    return value[len(prefix):]
+    raise Exception("Failed to determine OpenStack release")
+
+
+def validate(kolla_image_tags: KollaImageTags):
+    """Validate the kolla_image_tags variable."""
+    tag_var_re = re.compile(r"^[a-z0-9_]+$")
```
Comment (jovial) [inline tools/kolla-images.py:224]:
```suggestion
    tag_var_re = re.compile(r"^[a-z0-9_-]+$")
```
e.g haproxy-ssh

**Review by markgoddard — APPROVED** [review]
LGTM, but no CI to test :(

**Comment by jovial** [comment]
> LGTM, but no CI to test :(



> LGTM, but no CI to test :(

Thanks, for what it is worth all the non-upgrade jobs were passing: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/7762584458 in the run prior to https://github.com/stackhpc/stackhpc-kayobe-config/pull/909/commits/7b5f902f14a5fa39a89ab7677b8bef9af060b9fe. I'll wait a little bit to see if SMS comes back online before merging.

**Comment by markgoddard** [comment]
> > LGTM, but no CI to test :(
> 
> > LGTM, but no CI to test :(
> 
> Thanks, for what it is worth all the non-upgrade jobs were passing: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/7762584458 in the run prior to [7b5f902](https://github.com/stackhpc/stackhpc-kayobe-config/commit/7b5f902f14a5fa39a89ab7677b8bef9af060b9fe). I'll wait a little bit to see if SMS comes back online before merging.

I wouldn't hold your breath...

If the non-upgrade jobs passed let's merge.

**Pull request merged**